File: app/src/main/java/com/webviewblock/di/modules/FragmentBuildersModule.java
package com.webviewblock.di.modules;

import com.webviewblock.presentation.search.SearchFragment;
import com.webviewblock.presentation.settings.SettingsFragment;
import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class FragmentBuildersModule {
    @ContributesAndroidInjector
    abstract SearchFragment contributeSearchFragment();

    @ContributesAndroidInjector
    abstract SettingsFragment contributeSettingsFragment();
}

File: app/src/main/java/com/webviewblock/di/modules/FragmentBuildersModule.kt
package com.webviewblock.di.modules

import com.webviewblock.presentation.search.SearchFragment
import com.webviewblock.presentation.settings.SettingsFragment
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Suppress("unused")
@Module
abstract class FragmentBuildersModule {

    @ContributesAndroidInjector
    abstract fun contributeSearchFragment(): SearchFragment

    @ContributesAndroidInjector
    abstract fun contributeSettingsFragment(): SettingsFragment

//    @ContributesAndroidInjector
//    abstract fun contributeWebviewFragment(): WebviewFragment
}
